internal/host/conditions.go

```diff
@@ -0,0 +1,36 @@
+package host
+
+import (
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/models"
+)
+
+type conditionId string
+type condition struct {
+	id conditionId
+	fn func(c *validationContext) bool
+}
+
+const (
+	DiskSpeedCheckSuccessful = conditionId("disk-speed-check-successful")
+	DiskSpeedCheckFailed     = conditionId("disk-speed-check-failed")
+	ClusterInsufficient      = conditionId("cluster-insufficient")
+)
+
+func (c conditionId) String() string {
+	return string(c)
+}
+
+func (v *validator) isDiskSpeedCheckSuccessful(c *validationContext) bool {
```

Comment:
I would change naming to be **installation** disk, so it would be clear it's talking on the installation disk specifically and not any disk
`isInstallationDiskSpeedCheckSuccessful`